src/index.ts

```diff
@@ -24,11 +26,14 @@ export async function start(): Promise<void> {
   }>(webpack.filters.byProps("getChannel"));
 
   if (typingMod && getChannelMod) {
-    inject.instead(typingMod, "startTyping", ([channel], original) => {
-      if (cfg.get("invisible", true)) {
+    inject.instead(typingMod, "startTyping", ([channelId]: [string], original) => {
+      const global = cfg.get("invisible", true);
+      const channelWise = cfg.get("button", true) ? cfg.get("channelWise", true) : false;
+      const channels = cfg.get("channels", { [channelId]: global });
+      if (channelWise ? channels[channelId] ?? global : global) {
```

Comment:
don't use the global keyword. Use `_global` or something or a different var name.